src/OrchardCore/OrchardCore.Infrastructure/Validation/ValidatorWrapper.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Validation
+{
+    /// <summary>
+    /// Represents a wrapper class for <see cref="Validator"/> that can be used to validate objects.
+    /// </summary>
+    public static class ValidatorWrapper
+    {
+        /// <summary>
+        /// Determines whether the specified object is valid using the validation context and validation results collection.
+        /// </summary>
+        /// <param name="instance">The object to validate.</param>
+        /// <param name="validationContext">The context that describes the object to validate.</param>
+        /// <param name="validationResults">A collection to hold each failed validation.</param>
+        public static bool TryValidateObject(object instance, ValidationContext validationContext, ICollection<ValidationResult> validationResults)
+            => Validator.TryValidateObject(instance, validationContext, validationResults);
+
+        /// <summary>
+        /// Determines whether the specified object is valid using the validation context and validation results collection.
+        /// </summary>
+        /// <param name="instance">The object to validate.</param>
+        /// <param name="validationContext">The context that describes the object to validate.</param>
+        /// <param name="validationResults">A collection to hold each failed validation.</param>
+        public static async Task<bool> TryValidateObjectAsync(object instance, ValidationContext validationContext, ICollection<ValidationResult> validationResults)
```

Comment:
The API could instead be something like:

```suggestion
        public static async Task<bool> TryValidateModelAsync<TModel>(TModel instance, out ICollection<ValidationResult> validationResults)
```

Also, in controllers, this could be tied to `ModelState`.